lib/cn/cndb.c

```diff
@@ -132,22 +132,28 @@ cndb_info2omf(int hdr_type, const struct cndb_cn *cn, struct cndb_info_omf *inf)
 }
 
 merr_t
-cndb_alloc(struct mpool *ds, u64 *captgt, u64 *oid1_out, u64 *oid2_out)
+cndb_alloc(struct mpool *mp, u64 *captgt, u64 *oid1_out, u64 *oid2_out)
 {
-    merr_t            err;
-    enum mpool_mclass mclass;
-    size_t            capacity;
+    merr_t err;
+    size_t capacity;
+    int    i;
 
     if (captgt && *captgt)
         capacity = *captgt;
     else
         capacity = CNDB_CAPTGT_DEFAULT;
 
-    mclass = MP_MED_STAGING;
-    if (mpool_mclass_props_get(ds, mclass, NULL))
-        mclass = MP_MED_CAPACITY;
+    for (i = MP_MED_COUNT - 1; i >= MP_MED_BASE; i--) {
```

Comment:
Any particular reason you iterate backwards here?

Reply:
Yes, would like to check the availability of the best tier first.